src/Symfony/Component/HttpKernel/Tests/UriSignerTest.php

```diff
@@ -81,6 +114,61 @@ public function testSignerWorksWithFragments()
             'http://example.com/foo?_hash=EhpAUyEobiM3QTrKxoLOtQq5IsWyWedoXDPqIjzNj5o%3D&bar=foo&foo=bar#foobar',
             $signer->sign('http://example.com/foo?bar=foo&foo=bar#foobar')
         );
+
         $this->assertTrue($signer->check($signer->sign('http://example.com/foo?bar=foo&foo=bar#foobar')));
+
+        $this->assertSame(
+            'http://example.com/foo?_expiration=4070908800&_hash=qHl626U5d7LMsVtBxPt9GNzysdSxyOQ1fHA59Y1ib0Y%3D&bar=foo&foo=bar#foobar',
+            $signer->sign('http://example.com/foo?bar=foo&foo=bar#foobar', new \DateTime('2099-01-01 00:00:00'))
+        );
+
+        $this->assertTrue($signer->check($signer->sign('http://example.com/foo?bar=foo&foo=bar#foobar', new \DateTime('2099-01-01 00:00:00'))));
+    }
+
+    public function testSignWithUriExpiration()
+    {
+        $signer = new UriSigner('foobar');
+
+        $this->assertSame($signer->sign('http://example.com/foo?foo=bar&bar=foo', new \DateTime('2099-01-01 00:00:00')), $signer->sign('http://example.com/foo?bar=foo&foo=bar', 4070908800));
```

Comment:
please use `new DateTimeImmutable` unless the code is specifically testing support for the mutable class (which is not the case here).

We try to avoid the usage of the mutable value object as much as we can in Symfony in favor of the immutable one.